File: cpp/src/Position.cpp
#include <iostream>

#include "Position.hpp"

using namespace std;


/**
 * ---------------------------------------------------------------------------
 * Constructor
 * ---------------------------------------------------------------------------
 */

/**
 * @brief Default constructor that initializes both x and y to 0.
*/

Position::Position() : x(0), y(0) {}


/**
 * @brief Constructor that initializes the position with specified x and y coordinates.
 * @param ix The x-coordinate.
 * @param iy The y-coordinate.
*/

Position::Position(int ix, int iy) : x(ix), y(iy){}

/**
 * ---------------------------------------------------------------------------
*/

/**
 * @brief Get the x-coordinate of the position.
 * 
 * @return The x-coordinate.
*/

int Position::getX()
{
    return x;
}

/**
 * @brief Get the y-coordinate of the position.
 * 
 * @return The y-coordinate.
*/

int Position::getY()
{
    return y;
}


/**
 * @brief Set the x-coordinate of the position.
 * 
 * @param newX The new x-coordinate.
*/

void Position::setX(int newX)
{
    x = newX;
}

/**
 * @brief Set the y-coordinate of the position.
 * 
 * @param newY The new y-coordinate.
*/
void Position::setY(int newY)
{
    y = newY;
}

File: rust/src/position.rs
use super::*;

#[derive(Clone)]
pub struct Position {
    pub x: i32,
    pub y: i32,
}

impl Position {
    pub fn new(ix: i32, iy: i32) -> Self {
        Self { x: ix, y: iy }
    }

    pub fn default() -> Self {
        Self { x: 0, y: 0 }
    }

    pub fn get_x(&self) -> i32 {
        self.x
    }

    pub fn get_y(&self) -> i32 {
        self.y
    }

    pub fn set_x(&mut self, new_x: i32) {
        self.x = new_x;
    }

    pub fn set_y(&mut self, new_y: i32) {
        self.y = new_y;
    }
}